common/nymsphinx/src/preparer/mod.rs

```diff
@@ -192,9 +196,18 @@ pub trait FragmentPreparer {
         packet_type: PacketType,
         mix_hops: Option<u8>,
     ) -> Result<PreparedFragment, NymTopologyError> {
+        debug!("Preparing chunk for sending");
         // each plain or repliable packet (i.e. not a reply) attaches an ephemeral public key so that the recipient
         // could perform diffie-hellman with its own keys followed by a kdf to re-derive
         // the packet encryption key
+
+        let seed = fragment.seed().wrapping_mul(self.nonce());
+        let mut rng = ChaCha8Rng::seed_from_u64(seed as u64);
```

Comment:
I'm a bit afraid of rather low entropy of this `rng`, especially if "normal" client packets return constant `1` `nonce` -> could we make that one also return a proper pseudorandom value instead?